include/dgl/base_heterograph.h

```diff
@@ -547,6 +553,15 @@ DGL_DEFINE_OBJECT_REF(FlattenedHeteroGraphRef, FlattenedHeteroGraph);
 HeteroGraphPtr CreateHeteroGraph(
     GraphPtr meta_graph, const std::vector<HeteroGraphPtr>& rel_graphs);
 
+/*!
+ * \brief Create a heterograph from meta graph and a list of bipartite graph,
+ * additionally specifying number of nodes per type.
+ */
+HeteroGraphPtr CreateHeteroGraph(
+    GraphPtr meta_graph,
+    const std::vector<HeteroGraphPtr>& rel_graphs,
+    std::vector<int64_t> num_nodes_per_type);
```

Comment:
Let's have both of them be `std::vector` and also merge it with L553 by adding a default argument?

Reply:
If the second argument is pass-by-value then we will copy `rel_graphs` right?